examples/md-flexible/src/parsing/CLIParser.cpp

```diff
@@ -8,62 +8,72 @@
+      config.boxLength, config.containerOptions, config.cutoff, config.dataLayoutOptions, config.deltaT,
+      config.dontCreateEndConfig, config.tuningMaxEvidence, config.functorOption, config.dontMeasureFlops,
+      config.generatorOption, config.iterations, config.tuningInterval, config.logLevel, config.logFileName,
+      config.distributionMean, config.maxTuningPhasesWithoutTest, config.particlesPerDim, config.particlesTotal,
+      config.relativeOptimumRange, config.periodic, config.tuningPhases, config.verletClusterSize,
+      config.verletSkinRadius, config.particleSpacing, config.tuningSamples, config.traversalOptions,
+      config.tuningStrategyOption, config.useThermostat, config.verletRebuildFrequency, config.vtkFileName,
+      config.vtkWriteFrequency, config.selectorStrategy, config.yamlFilename, config.distributionStdDev,
+      MDFlexConfig::MDFlexOption<std::string, 'Z'>("", "zsh-completions", false, "Generate completions file for zsh."),
+      MDFlexConfig::MDFlexOption<std::string, 'h'>("", "help", false, "Display this message."))};
+
+  constexpr auto relevantOptionsSize = std::tuple_size_v<decltype(relevantOptions)>;
+
+  // sanity check that all getopt chars are unique. Brackets for scoping.
+  {
+    // map tracking mappings of getopt chars to strings
+    std::map<char, std::string> getoptCharsToName;
+    // look for clashes by checking if getopt chars are in the map and otherwise add them
+    autopas::utils::TupleUtils::for_each(relevantOptions, [&](auto &opt) {
+      if (auto iterAtClash = getoptCharsToName.find(opt.getoptChar); iterAtClash != getoptCharsToName.end()) {
+        throw std::runtime_error("CLIParser::parseInput: the following options share the same getopt char!\n" +
+                                 opt.name + "\n" + iterAtClash->second);
+      } else {
+        getoptCharsToName.insert({opt.getoptChar, opt.name});
+      }
+    });
+  }
+
+  // create data structure for options that getopt can use
+  std::vector<struct option> long_options;
+  // reserve space for all relevant options and terminal field
+  long_options.reserve(relevantOptionsSize + 1);
+
+  autopas::utils::TupleUtils::for_each(relevantOptions,
+                                       [&](auto &elem) { long_options.push_back(elem.toGetoptOption()); });
+
+  // needed to signal the end of the array
+  long_options.push_back({nullptr, no_argument, nullptr, 0});
+
   // reset getopt to scan from the start of argv
   optind = 1;
   bool displayHelp = false;
   for (int cliOption = 0, cliOptionIndex = 0;
-       (cliOption = getopt_long(argc, argv, "", long_options, &cliOptionIndex)) != -1;) {
+       (cliOption = getopt_long(argc, argv, "", long_options.data(), &cliOptionIndex)) != -1;) {
     string strArg;
     if (optarg != nullptr) strArg = optarg;
     transform(strArg.begin(), strArg.end(), strArg.begin(), ::tolower);
     switch (cliOption) {
       case '3': {
```

Comment:
I still would like to have a variable here instead of the literal. However, I currently see only three options and I like none of them ;) 
1. make all option members of MDFlexConfig static, then we could write something like:
```C++
case MDFlexConfig::newton3option.getoptChar() :
```
Currently this is not possible because newton3option is not available at compile time.
2.  make constexpr variables for all `getoptChars` next to the option members. So in MDFlexConfig.h something like:
```C++
  constexpr static char newton3OptionsGetoptChar{'3'};
  /**
   * newton3Options
   */
  MDFlexOption<std::set<autopas::Newton3Option>, newton3OptionsGetoptChar> newton3Options{
      autopas::Newton3Option::getAllOptions(), "newton3", true,
      "List of newton3 options to use. Possible Values: " +
          autopas::utils::ArrayUtils::to_string(autopas::Newton3Option::getAllOptions(), " ", {"(", ")"})};
```
then we can here use `case newton3OptionsGetoptChar: `
3. replacing the `switch` with a large `if else` tree

Reply:
fixed via `decltype(option)::getoptChar` in d52640c2